Lib/test/test_grammar.py

```diff
@@ -991,6 +991,29 @@ def g3():
                 return 4
         self.assertEqual(g3(), 4)
 
+    def test_break_in_finally_after_return(self):
+        # See issue #37830
+        def test():
+            for i in [0, 1]:
+                for j in [10, 20]:
+                    try:
+                        return i + j
+                    finally:
+                        break
```

Comment:
Could we add some versions of these tests that have a conditional `break` and `continue` in the `finally`? These cases are not covered anywhere else in the test suite and other simple approaches for this usually fail in them

Reply:
Done.